xdrip-get-entries.sh

```diff
@@ -1060,25 +1078,28 @@ function  capture_entry_values()
   sessionStartDate="${sessionStartDate%\"}"
   sessionStartDate="${sessionStartDate#\"}"
   sessionStartDateEpochms=$(cat ${LDIR}/extra.json | jq -M '.[0].sessionStartDateEpoch')
-  sessionMinutesRemaining=$(bc <<< "($SECONDS_IN_10_DAYS - ($epochdate-$sessionStartDateEpochms/1000))/60")
-  # EYF here
-  # TODO make sure to use 7 days for g5 and 10 for g6
-  # check for valid and not expired sessionStartDate
+  # make sure to use 7 days for g5 and 10 for g6
+  local sessionMaxSeconds=$SECONDS_IN_10_DAYS
   if [ "$txType" == "g5" ]; then
-    :
+    sessionMaxSeconds=$SECONDS_IN_7_DAYS
   fi
 
-  if [ $(bc <<< "$sessionMinutesRemaining < 0") -eq 1 -a $(bc <<< "$sessionMinutesRemaining > ($SECONDS_IN_10_DAYS * 60)") -eq 1 ]; then
+  # check for valid and not expired sessionStartDate
+  sessionMinutesRemaining=$(bc <<< "($sessionMaxSeconds - ($epochdate-$sessionStartDateEpochms/1000))/60")
+  if [ $(bc <<< "$sessionMinutesRemaining < 0") -eq 1 -a $(bc <<< "$sessionMinutesRemaining > ($sessionMaxSeconds * 60)") -eq 1 ]; then
     log "Expired session or invalid sessionStartDate, not processing auto-restart logic"
-    sessionMinutesRemaining=100000 # big number ensures auto-restart logic will not kick in
   fi
   log "sessionStartDate=$sessionStartDate, sessionStartDateEpochms=$sessionStartDateEpochms" 
   log "sessionMinutesRemaining=$sessionMinutesRemaining"
   if [ $(bc <<< "$sessionMinutesRemaining < 65") -eq 1 ]; then
     #TODO use parameter to do this only if auto-restart is true
     if [ $(bc <<< "$glucose < 400") -eq 1  -a $(bc <<< "$glucose > 40") -eq 1 ]; then
       if [ $(bc <<< "$variation < 10") -eq 1 ]; then
+        if [[ "$auto_sensor_restart" == true ]]; then
          cgm-stop; sleep 1; cgm-start -m 120; sleep 1; cgm-calibrate $glucose; sleep 1; cgm-calibrate $glucose  
```

Comment:
this is amazingly cool if it works!

Reply:
It worked for me at least once already. It takes a couple of cycles for the tx to say OK, but it did work.